docs/knative-rabbitmq.md

```diff
@@ -0,0 +1,335 @@
+```
+
+## Knative broker
+
+### Create broker config
+
+```yaml
+# broker-config.yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: RabbitmqBrokerConfig
+metadata:
+  name: broker-config
+spec:
+  rabbitmqClusterReference:
+    # Configure name if a RabbitMQ Cluster Operator is being used.
+    name: rabbitmq-cluster
+  queueType: quorum
+```
+
+```sh
+kubectl apply -f broker-config.yaml
+```
+
+### Create broker
+
+```yaml
+# broker.yaml
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  annotations:
+    eventing.knative.dev/broker.class: RabbitMQBroker
+  name: broker
+spec:
+  config:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: RabbitmqBrokerConfig
+    name: broker-config
+```
+
+```sh
+kubectl apply -f broker.yaml
+```
+
+## Knative trigger
+
+```yaml
+# trigger.yaml
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: my-service-trigger
+spec:
+  broker: broker
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: hello
+```
+
+```
+kubectl apply -f trigger.yaml
+```
+
+## Deploy hello example function
+
+```
+func create -l go hello
+```
+
+modify handle.go
+
+```go
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Handle an HTTP Request.
+func Handle(ctx context.Context, res http.ResponseWriter, req *http.Request) {
+	/*
+	 * YOUR CODE HERE
+	 *
+	 * Try running `go test`.  Add more test as you code in `handle_test.go`.
+	 */
+
+	fmt.Println("Received request")
+	fmt.Println(prettyPrint(req))      // echo to local output
+	fmt.Fprintf(res, prettyPrint(req)) // echo to caller
+}
+
+func prettyPrint(req *http.Request) string {
+	b := &strings.Builder{}
+	fmt.Fprintf(b, "%v %v %v %v\n", req.Method, req.URL, req.Proto, req.Host)
+	for k, vv := range req.Header {
+		for _, v := range vv {
+			fmt.Fprintf(b, "  %v: %v\n", k, v)
+		}
+	}
+
+	if req.Method == "POST" {
+		// Remove req.ParseForm()
+
+		fmt.Fprintln(b, "Body:")
+		bodyBytes, err := io.ReadAll(req.Body)
+		if err != nil {
+			fmt.Fprintf(b, "Error reading body: %v", err)
+			return b.String()
+		}
+
+		var eventData map[string]interface{}
+		if err := json.Unmarshal(bodyBytes, &eventData); err != nil {
+			fmt.Fprintf(b, "Error parsing JSON: %v", err)
+		} else {
+			for k, v := range eventData {
+				fmt.Fprintf(b, "  %v: %v\n", k, v)
+			}
+		}
+	}
+
+	return b.String()
+}
+```
+
+```sh
+cd hello
+```
+
+```
+func deploy
+```
+
+enter your registry (usually docker.io/<your_username>)
+
+### list knative service
+
+```
+kubectl get ksvc
+```
+
+you should see the hello service just created
+
+## Install kn event plugin
+
+https://github.com/knative-extensions/kn-plugin-event
+
+```sh
+kubectl get pods -w
+```
+
+Open another terminal then run
+
+```sh
+kn event send \
+  --to Broker:eventing.knative.dev/v1:broker
+```
+
+You should see our knative has created the pod to handle the trigger from event
```

Comment:
Did you try the rabbitMQ plugin to: emit message from rabbitMQ and trigger knative call to a function?

Reply:
I've tried rabbitmqadmin publish like `rabbitmqadmin publish exchange=amq.default routing_key=<trigger_name> payload="Hello rabbitmq" --username=<username> --password=<pwd>`